crayfish/processing/parameters.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+
+# Crayfish - A collection of tools for TUFLOW and other hydraulic modelling packages
+# Copyright (C) 2016 Lutra Consulting
+
+# info at lutraconsulting dot co dot uk
+# Lutra Consulting
+# 23 Chestnut Close
+# Burgess Hill
+# West Sussex
+# RH15 8HN
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import datetime
+
+from processing.gui.wrappers import EnumWidgetWrapper, InvalidParameterValue
+from qgis.core import QgsProcessingParameterEnum, QgsMeshDatasetIndex
+
+
+class DatasetGroupType():
+    DataOnFaces = 0
+    DataOnVertices = 1
+
+
+class DatasetWrapper(EnumWidgetWrapper):
+
+    def on_change(self, wrapper):
+        mesh_layer = wrapper.widgetValue()
+        if mesh_layer:
+            dp = mesh_layer.dataProvider()
+            dataset_group_filter = self.parameterDefinition().datasetGroupFilter
+
+            options = []
+            for i in range(dp.datasetGroupCount()):
+                meta = dp.datasetGroupMetadata(i)
+                if not dataset_group_filter or dataset_group_filter(meta):
+                    options.append(meta.name())
+        else:
+            options = []
+        self.parameterDefinition().setOptions(options)
+        if self.parameterDefinition().allowMultiple():
+            self.widget.updateForOptions(options)
+        else:
+            self.widget.clear()
+            for i, opt in enumerate(options):
```

Comment:
this is not right, since you filter out groups based on if they are defined on faces/vertices. so you need to use map, not vector for groupIndex -> groupName